step-1

DubiumEkb opened the pull request.

DubiumEkb committed:
step-1

gennady-bars commented on src/components/BurgerConstructor/BurgerConstructor.tsx:
```diff
@@ -0,0 +1,35 @@
+// Import Style
+import style from "./BurgerConstructor.module.css"
+
+const BurgerConstructor = ({data}:any) => {
```
`({data}:any) ` тип `any` нельзя использовать в пропсах. Либо нужно типизировать пропсы с помощью `typescript`, либо убрать расширение `ts` из файлов и типизировать пропсы с помощью `propTypes`

gennady-bars commented on src/components/BurgerIngredients/BurgerIngredients.tsx:
```diff
@@ -0,0 +1,73 @@
+					<div className={style.BurgerIngredientsContentSectionRow}>
+						{data.map((item:any) => {
+							return item.type === "bun" ? (
+								<BurgerIngredientsItem key={item._id} item={item} classname="pb-10" />
```
`key={item._id} `

Отлично, что используете атрибут `key`, и отлично, что для него используете уникальный `_id`, а не индекс массива.

[Почему индекс не нужно использовать, можно прочитать тут]( https://medium.com/russian/%D0%B8%D0%BD%D0%B4%D0%B5%D0%BA%D1%81-%D0%B2-%D0%BA%D0%B0%D1%87%D0%B5%D1%81%D1%82%D0%B2%D0%B5-%D0%BA%D0%BB%D1%8E%D1%87%D0%B0-%D0%B0%D0%BD%D1%82%D0%B8-%D0%BF%D0%B0%D1%82%D1%82%D0%B5%D1%80%D0%BD-56bab8ada309)

gennady-bars reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Работа проделана огромная


* Отличная верстка
* Отлично, что используете атрибут `key`, и отлично, что для него используете уникальный `_id`, а не индекс массива.

но есть некоторые недочеты:


* Файл `App.test.js` нужно удалить, так как он не относится к проекту
* ` App.tsx` нужно переместить в папку `components` и там поместить в свою папку `App`
* `App.module.css` нужно переместить в папку к `App.tsx` 
* Скрин http://joxi.ru/1A5YdBpS464O1A Внутри заказа не должно быть **других булок**: есть только верхняя и нижняя одинаковые.
* у ингредиентов между булками должна быть иконка удаления (мусорный бачок). Скрин http://joxi.ru/vAW8b9ZT3x3Xyr 
* `({data}:any) ` тип `any` нельзя использовать в пропсах. Либо нужно типизировать пропсы с помощью `typescript`, либо убрать расширение `ts` из файлов и типизировать пропсы с помощью `propTypes` . Это нужно исправить во всех компонентах
* Нужно посмотреть все компоненты и типизировать их `пропсы`, если где-то пропустили. 


Исправьте, пожалуйста, недочеты и работа будет принята. Пожалуйста, проверьте работоспособность проекта и наличие возможных ошибок в консоли браузера (кнопка `F12`) перед отправкой на ревью. 

Напоминаю, что работа может быть принята только после исправления всех критических замечаний `Нужно исправить`.

Удачного рефакторинга кода.

DubiumEkb committed:
deleted:    src/App.test.tsx
	renamed:    src/App.module.css -> src/components/App/App.module.css
	renamed:    src/App.tsx -> src/components/App/App.tsx
	modified:   src/components/BurgerConstructor/BurgerConstructor.tsx
	modified:   src/components/BurgerConstructor/ui/BurgerConstructorItem/index.tsx
	modified:   src/components/BurgerIngredients/BurgerIngredients.tsx
	modified:   src/index.tsx
	new file:   src/utils/dataType.ts

gennady-bars reviewed (approved):
Поздравляю! Ваша работа принята.

Вы отлично потрудились. Хорошая типизация пропсов

Удачного дальнейшего обучения.


p.s. теперь нужно сделать 2й этап и прислать в этом же тикете
